src/stores/ResourceConfig/Store.ts

```diff
@@ -460,33 +463,59 @@ const getInitialState = (
                 liveBackfillCounter !== draftedBackfillCounter ||
                 (liveBindingIndex === -1 && draftedBackfillCounter > 0)
             ) {
-                addBackfilledCollection(collection);
+                addBackfilledCollections([collection]);
             }
         });
     },
 
-    addBackfilledCollection: (value) => {
+    addBackfilledCollections: (values) => {
         set(
             produce((state: ResourceConfigState) => {
-                if (!state.backfilledCollections.includes(value)) {
-                    state.backfilledCollections.push(value);
-                }
+                state.backfilledCollections = union(
+                    state.backfilledCollections,
+                    values
+                );
             }),
             false,
-            'Backfilled Collection Added'
+            'Backfilled Collections Added'
         );
     },
 
-    removeBackfilledCollection: (value) => {
+    removeBackfilledCollections: (values) => {
         set(
             produce((state: ResourceConfigState) => {
                 state.backfilledCollections =
                     state.backfilledCollections.filter(
-                        (collection) => collection !== value
+                        (collection) => !values.includes(collection)
                     );
             }),
             false,
-            'Backfilled Collection Removed'
+            'Backfilled Collections Removed'
+        );
+    },
+
+    setBackfilledCollections: (increment, targetCollection) => {
+        set(
+            produce((state: ResourceConfigState) => {
+                const collections: string[] = targetCollection
+                    ? [targetCollection]
+                    : state.collections
+                    ? state.collections
+                    : [];
+
+                state.backfilledCollections =
+                    increment === 'true'
+                        ? union(state.backfilledCollections, collections)
```

Comment:
Unless we need the sorting - I don't think we need to use `union` here. I think `concat` would be okay.